src/sage/modules/free_module_pseudohomspace.py

```diff
@@ -0,0 +1,262 @@
+from sage.rings.polynomial.ore_polynomial_ring import OrePolynomialRing
+from sage.modules.free_module_pseudomorphism import FreeModulePseudoMorphism
+
+
+class FreeModulePseudoHomspace(UniqueRepresentation, HomsetWithBase):
+    r"""
+    This class implements the space of Pseudomorphisms with a fixed twist.
+
+    For free modules, the elements of a pseudomorphism correspond to matrices
+    which define the mapping on elements of a basis.
+
+    TESTS::
+
+        sage: F = GF(125)
+        sage: M = F^2
+        sage: Frob = F.frobenius_endomorphism()
+        sage: PHS = M.pseudoHom(Frob)
+        sage: h = PHS([[1, 2], [1, 1]])
+        sage: e = M((4*F.gen()^2 + F.gen() + 2, 4*F.gen()^2 + 4*F.gen() + 4))
+        sage: h(e)
+        (z3, 2*z3^2 + 3*z3 + 3)
+    """
+    Element = FreeModulePseudoMorphism
+
+    @staticmethod
+    def __classcall_private__(cls, domain, codomain, twist):
+        r"""
+        Constructs the space of pseudomorphisms with a given twist.
+
+        INPUT:
+
+        -  ``domain`` -- a free module,  the domain of this pseudomorphism
+
+        -  ``codomain`` -- a free module, the codomain of this pseudomorphism
+
+        -  ``twist`` -- a twisting morphism/derivation or a Ore polynomial ring
+
+        TESTS::
+
+            sage: F = GF(125)
+            sage: Frob = F.frobenius_endomorphism()
+            sage: M = F^2
+            sage: H = M.pseudoHom(Frob)
+            sage: type(H)
+            <class 'sage.modules.free_module_pseudohomspace.FreeModulePseudoHomspace_with_category'>
+
+            sage: # Testsuite(H).run()
+        """
+        ring = domain.base_ring()
+        if codomain.base_ring() is not ring:
+            raise ValueError("the domain and the codomain must be defined over the same ring")
+        if isinstance(twist, OrePolynomialRing):
+            ore = twist
+            if ore.base_ring() is not ring:
+                raise ValueError("base rings do not match")
+        else:
+            ore = OrePolynomialRing(ring, twist, names='x', polcast=False)
+        if ore._derivation is not None:
+            if not codomain.has_coerce_map_from(domain):
+                raise ValueError("the domain does not coerce into the codomain")
+        return cls.__classcall__(cls, domain, codomain, ore)
+
+    def __init__(self, domain, codomain, ore):
+        r"""
+        Initialize this pseudohom space.
+
+        INPUT:
+
+        -  ``domain`` -- a free module,  the domain of this pseudomorphism
+
+        -  ``codomain`` -- a free module, the codomain of this pseudomorphism
+
+        -  ``ore`` -- the underlying Ore polynomial ring
+
+        TESTS::
+
+            sage: F = GF(125)
+            sage: Frob = F.frobenius_endomorphism()
+            sage: M = F^2
+            sage: M.pseudoHom(Frob)
+            Set of Pseudoendomorphisms (twisted by z3 |--> z3^5) of Vector space of dimension 2 over Finite Field in z3 of size 5^3
+        """
```

Comment:
You may wrap long output. I would cut at "over Finite ..."

Reply:
I cutted just before "Vector space". I prefer, I think.